endpoint_url: "opc.tcp://localhost:4840"
variables:
  Point:
    description: "This is a description"
    type: struct
    index: 1
    typeID: "{1, 1}" # example
    bynaryTypeID: "{1, 4042 }" # example
    elements:
      - { type: float, name: x } # datatype and names
      - { type: float, name: y }
      - { type: float, name: z }

  Measurements:
    description: "This is a description"
    type: struct
    index: 1
    typeID: "{1, 2}" # example
    bynaryTypeID: "{1, 4443 }" # example
    elements:
      - { type: opcua::String, name: description } # datatype and names
      - { type: size_t, name: mesurementSize }
      - { type: float*, name: measurements}

  Opt:
    description: "Some description"
    type: struct
    index: 1
    typeID: "{1, 3}"
    bynaryTypeID: "{1, 4644}"
    elements:
      - { type: int16_t, name: a }
      - { type: float*, name: b }
      - { type: float*, name: c }

  variable3:
    description: "This is a description"
    type: int

  variable4:
    description: "This is a description"
    type: bool

  variable5:
    description: "This is a description"
    type: string
